### PR TITLE
@broskoski: Delete keys with requests that 404

### PR DESCRIPTION
**Pull request by dzucconi**
Slightly more useful error handling as well for API errors (by including the failed URL):

### CODE REVIEW

File: lib/apis/fetch.js [hunk 1]
```diff
@@ -11,7 +11,10 @@ export default (url, options = {}) => {
 
     request(url, opts, (err, response) => {
       if (!!err || response.statusCode !== 200) {
-        return reject(err || new Error(response.body));
+        if (err) return reject(err);
+        const error = new Error(`${response.request.uri.href}: ${response.body}`);
```
Comment (broskoski) [inline lib/apis/fetch.js]: is it safe to assume that all these properties (`response.request.uri.href`) will always exist?

Comment (dzucconi) [inline lib/apis/fetch.js]: I believe it is, but I just updated the code to assume nothing.

File: lib/apis/fetch.js [hunk 2]
```diff
@@ -11,7 +11,15 @@ export default (url, options = {}) => {
...
     request(url, opts, (err, response) => {
       if (!!err || response.statusCode !== 200) {
-        return reject(err || new Error(response.body));
+        if (err) return reject(err);
+
+        const message = compact([
+          get(response, 'request.uri.href'),
+          response.body,
+        ]).join(' - ');
+        const error = new Error(message);
```
Comment (broskoski) [inline lib/apis/fetch.js]: 🌞

File: lib/apis/fetch.js [hunk 3]
```diff
@@ -11,7 +11,15 @@ export default (url, options = {}) => {
 
     request(url, opts, (err, response) => {
       if (!!err || response.statusCode !== 200) {
-        return reject(err || new Error(response.body));
+        if (err) return reject(err);
```
Comment (broskoski) [inline lib/apis/fetch.js]: What's the case where this will not just return a rejection? When would `err` be null?

Comment (dzucconi) [inline lib/apis/fetch.js]: Some errors use the callback; some don't. In this case we'll get valid responses which are 4xx errors; so we just check the status code instead. I agree: it's weird.

**Comment by broskoski** [comment]
Ok, merging. ✌️

**Pull request merged**